RetailCoder.VBE/Refactorings/ExtractMethod/ExtractMethodModel.cs

```diff
@@ -165,55 +129,62 @@ public string PreviewCode
+                            _parametersList.Add(parameter.ToString(ExtractMethodParameterFormat.DimOrParameterDeclarationWithAccessibility));
                             break;
-                        case ExtractParameterNewType.PrivateLocalVariable:
-                            variables.Add(string.Format($"{Tokens.Dim} {parameter.Name} {Tokens.As} {parameter.TypeName}"));
+                        case ExtractMethodParameterType.PrivateLocalVariable:
+                            _variablesList.Add(parameter.ToString(ExtractMethodParameterFormat.DimOrParameterDeclarationWithAccessibility));
                             break;
-                        case ExtractParameterNewType.StaticLocalVariable:
-                            variables.Add(string.Format($"{Tokens.Static} {parameter.Name} {Tokens.As} {parameter.TypeName}"));
+                        case ExtractMethodParameterType.StaticLocalVariable:
+                            _variablesList.Add(parameter.ToString(ExtractMethodParameterFormat.DimOrParameterDeclarationWithAccessibility));
                             break;
                         default:
                             throw new InvalidOperationException("Invalid value for ExtractParameterNewType");
                     }
                 }
 
-                var isFunction = ReturnParameter != null;
+                var isFunction = ReturnParameter != null &&
+                                 !(ReturnParameter.TypeName == string.Empty &&
+                                   ReturnParameter.Name == RubberduckUI.ExtractMethod_NoneSelected);
 
                 /* 
                    string.Empty are used to create blank lines
                    as the joins will create a newline each line.
                 */
 
                 var strings = new List<string>();
-                if (fields.Any())
+                if (_fieldsList.Any())
                 {
-                    strings.AddRange(fields);
+                    strings.AddRange(_fieldsList);
                     strings.Add(string.Empty);
                 }
-                strings.Add($@"{Tokens.Private} {(isFunction ? Tokens.Function : Tokens.Sub)} {NewMethodName ?? "NewMethod"}({string.Join(", " , parameters)}) {(isFunction ? string.Concat(Tokens.As, " ", ReturnParameter.TypeName ?? Tokens.Variant) : string.Empty)}");
-                strings.AddRange(variables);
-                if (variables.Any())
+                strings.Add(
+                    $@"{Tokens.Private} {(isFunction ? Tokens.Function : Tokens.Sub)} {
+                            NewMethodName ?? RubberduckUI.ExtractMethod_DefaultNewMethodName
+                        }({string.Join(", ", _parametersList)}) {
+                            (isFunction
+                                ? string.Concat(Tokens.As, " ", ReturnParameter.ToString(ExtractMethodParameterFormat.ReturnDeclaration) ?? Tokens.Variant)
+                                : string.Empty)
+                        }");
```

Comment:
this one seems somewhat hard to follow. maybe it's better with cleaner syntax-highlighting...